### ISSUE TITLE
some CLI command do not work with typescript

### ISSUE THREAD

**Issue by DimitriGilbert** [strapi/strapi#13095]
### Required System information

- Node.js version: 16.13.2
- NPM version: 8.2
- Strapi version: 4.2.0-beta0
- Database: mysql
- Operating system: ubuntu

### Describe the bug

When using typescript, some commands do not seem to be using the ```dist/``` folder.
So far I tested the following commands:
* console
* config:dump
* admin:reset-password
* admin:create
* routes:list

### Steps to reproduce the behavior

1. open a terminal in your typescript project
2. run ```yarn strapi console```
3. See error

### Expected behavior

A usable CLI

### Additional context

See bug [#13079](https://github.com/strapi/strapi/issues/13079)

**Comment by DimitriGilbert** [comment]
Example fix for [commands/console](github.com/strapi/strapi/blob/592ecb362eadb29e5c117879a8723b18b796df38/packages/core/strapi/lib/commands/console.js)

```javascript
'use strict';

const path = require('path');
const REPL = require('repl');
const strapi = require('../index');
const tsUtils = require('@strapi/typescript-utils');

/**
 * `$ strapi console`
 */
module.exports = async () => {
  const appDir = process.cwd();

  const isTSProject = await tsUtils.isTypeScriptProject(appDir);
  const distDir = isTSProject ? path.join(appDir, 'dist') : appDir;
  // Now load up the Strapi framework for real.
  const app = strapi({ distDir, appDir });

  app.start().then(() => {
    const repl = REPL.start(app.config.info.name + ' > ' || 'strapi > '); // eslint-disable-line prefer-template

    repl.on('exit', function(err) {
      if (err) {
        app.log.error(err);
        process.exit(1);
      }

      app.server.destroy();
      process.exit(0);
    });
  });
};

```

**Comment by Convly** [comment]
Hey, it should be fixed in v4.2.0-beta.2, I'll close it for now but don't hesitate to reopen it if you're still having the issue.
Have a nice day!
